lib/commands/sdk.js

```diff
@@ -549,427 +542,505 @@ SdkSubcommands.install = {
+						resolve(tempName);
 					});
 				}
 			});
-		}
-
-		function extractSDK(filename, keepFiles, version, callback) {
-			logger.log(__('Extracting SDK to %s', installLocation.cyan));
-
-			var bar;
+		});
+	}
 
-			appc.zip.unzip(filename, installLocation, {
-				visitor: function (entry, i, total) {
-					if (i === 0 && !cli.argv.quiet && !!cli.argv['progress-bars']) {
+	/**
+	 * @param {string} installLocation destination folder
+	 * @returns {Promise<void>}
+	 */
+	async extract(installLocation) {
+		this.logger.log(__('Extracting SDK to %s', installLocation.cyan));
+
+		return new Promise((resolve, reject) => {
+			let bar;
+			appc.zip.unzip(this.file, installLocation, {
+				visitor: (entry, i, total) => {
+					if (i === 0 && this._showProgress) {
 						bar = new appc.progress('  :paddedPercent [:bar]', {
 							complete: '='.cyan,
 							incomplete: '.'.grey,
 							width: 40,
-							total: total
+							total
 						});
 					}
 
 					bar && bar.tick();
 
-					if (!version) {
-						var m = entry.fileName.match(/^mobilesdk\/[^/]+\/([^/]+)\/(version\.txt|manifest\.json)$/);
+					// record the version if we don't have one set
+					if (!this.version) {
+						const m = entry.fileName.match(/^mobilesdk\/[^/]+\/([^/]+)\/(version\.txt|manifest\.json)$/);
 						if (m) {
-							version = m[1];
+							this.version = m[1];
 						}
 					}
 				}
-			}, function (err, extracted, total) {
+			}, (err, extracted, total) => {
+				// erase the zipfile if we've been asked to
+				if (!this.keepFiles) {
+					fs.unlinkSync(this.file);
+				}
+
 				if (err) {
-					keepFiles || fs.unlinkSync(filename);
-					logger.log();
-					if (version) {
-						logger.error(__('Failed to unzip Titanium SDK %s', version));
+					let str;
+					if (this.version) {
+						str = __('Failed to unzip Titanium SDK %s', this.version);
 					} else {
-						logger.error(__('Failed to unzip Titanium SDK'));
+						str = __('Failed to unzip Titanium SDK');
 					}
-					String(err).trim().split('\n').forEach(logger.error);
-					logger.log();
-					process.exit(1);
-				} else {
-					if (bar) {
-						bar.tick(total);
-						logger.log('\n');
-					}
-					keepFiles || fs.unlinkSync(filename);
+					str += String(err);
+					return reject(new Error(str));
+				}
 
-					// get the actual version number
-					var manifestFile = path.join(installLocation, 'manifest.json');
-					if (fs.existsSync(manifestFile)) {
-						var manifest = require(manifestFile);
-						version = manifest.version || version;
-					}
+				if (bar) {
+					bar.tick(total);
+					this.logger.log('\n');
+				}
 
-					callback(version);
+				// get the actual version number
+				const manifestFile = path.join(installLocation, 'manifest.json');
+				if (fs.existsSync(manifestFile)) {
+					const manifest = fs.readJSONSync(manifestFile);
+					this.version = manifest.version || this.version;
 				}
+
+				resolve();
 			});
+		});
+	}
+
+	/**
+	 * @param {object} sdks
```

Comment:
<!--
  0 failure: 
  1 warning:  lib/commands/sdk....
  
  
  DangerID: danger-id-Danger;
  File: lib/commands/sdk.js;
  Line: 785;
-->

- :warning: lib/commands/sdk.js line 785 – Missing JSDoc parameter description for 'sdks'. (valid-jsdoc)